hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -126,6 +126,10 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(@Nullable Object other) {
+		if (!(other instanceof NavigablePath)) {
```

Comment:
This is not really correct, because AFAICT from the code below, this method also considers a `NavigablePath` equal to a `DotIdentifierSequence` as long as local names and the parent match.

Reply:
that sounds like violation of contract, but since everyone is happy with that, I will revert